launcher/SysInfo.cpp

```diff
@@ -0,0 +1,58 @@
+#include <QString>
+#ifdef Q_OS_MACOS
+#include <sys/sysctl.h>
+#endif
+
+#ifdef Q_OS_MACOS
+bool rosettaDetect() {
+    int ret = 0;
+    size_t size = sizeof(ret);
+    if (sysctlbyname("sysctl.proc_translated", &ret, &size, NULL, 0) == -1)
+    {
+        return false;
+    }
+    if(ret == 0)
+    {
+        return false;
+    }
+    if(ret == 1)
+    {
+        return true;
+    }
+    return false;
+}
+#endif
+
+namespace SysInfo {
+    QString currentSystem() {
+#if defined(Q_OS_LINUX)
+        return "linux";
+#elif defined(Q_OS_MACOS)
+        return "osx";
+#elif defined(Q_OS_WINDOWS)
+        return "windows";
+#elif defined(Q_OS_FREEBSD)
+        return "freebsd";
+#elif defined(Q_OS_OPENBSD)
+        return "openbsd";
+#else
+        return "unknown";
+#endif
+    }
+
+    QString currentArch() {
+        auto qtArch = QSysInfo::currentCpuArchitecture();
+#ifdef Q_OS_MACOS
+        if(rosettaDetect())
```

Comment:
@dada513 this part is problematic. Since I'm compiling on my M1, it doesn't use translation and therefore the architecture is set to `x86_64`.

Reply:
Does `currentCpuArchitecture` alone not work? It seems like the documentation indicates that it would have the desired behavior here, and it seems to be clearly differentiated from [`buildCpuArchitecture`](https://doc.qt.io/qt-5.15/qsysinfo.html#buildCpuArchitecture). I don't see a reason to do anything special for Rosetta or macOS, unless this documentation isn't correct.

Reply:
it did not work last time I tried so I had to hack around with it.